lib/rules/selector-nested-pattern/index.mjs

```diff
@@ -49,20 +52,36 @@ const rule = (primary, secondaryOptions) => {
 				return;
 			}
 
-			const selectors = splitList ? ruleNode.selectors : [ruleNode.selector];
+			if (splitList) {
+				parseSelector(getRuleSelector(ruleNode), result, ruleNode)?.each((selector) => {
```

Comment:
```suggestion
			const selectors = getRuleSelector(ruleNode);

			if (splitList) {
				parseSelector(selectors, result, ruleNode)?.each((selector) => {
```

How about storing `getRuleSelector(ruleNode)` into a variable and re-using that in both branches?

```js
			} else {
				if (normalizedPattern.test(selectors)) {
					return;
				}

				report({
					result,
					ruleName,
					message: messages.expected,
					messageArgs: [selectors, primary],
					node: ruleNode,
					index: 0,
					endIndex: selectors.length
				});
			}
```

Reply:
You could also add a `check()` function to reduce duplication.

Something like :

```js
/**
 * @param {string} selectorStr
 * @param {number} index
 * @param {number} endIndex
 */
function check(selectorStr, index, endIndex) {
	if (normalizedPattern.test(selectorStr)) return;

	report({
		result,
		ruleName,
		message: messages.expected,
		messageArgs: [selectorStr, primary],
		node: ruleNode,
		index,
		endIndex,
	});
}
```

Reply:
Thanks for the feedback! I applied your suggestion.
[apply suggestions from code review](https://github.com/stylelint/stylelint/pull/8072/commits/6076e57cea3f061cf86868cf8f9a35284d34ea99)